src/libraries/Common/src/Internal/Cryptography/HashProviderCng.cs

```diff
@@ -26,7 +27,7 @@ public HashProviderCng(string hashAlgId, byte[]? key) : this(hashAlgId, key, isH
         {
         }
 
-        internal HashProviderCng(string hashAlgId, ReadOnlySpan<byte> key, bool isHmac)
+        internal HashProviderCng(string? hashAlgId, ReadOnlySpan<byte> key, bool isHmac)
```

Comment:
I don't think this accepts/expects null.

Reply:
Nullability here stems from `HashAlgorithmName.Name`, but you're right, it looks like non-nullability is being asserted in all callsites.